### PR TITLE
RATEST-159: Refactoring register patient test

### PR DESCRIPTION
**Pull request by kdaud**
Ticket [ID](https://issues.openmrs.org/browse/RATEST-159)
Description -> Refactoring register patient test

### CODE REVIEW

File: ui-tests/src/test/java/org/openmrs/reference/RegisterPatientTest.java [hunk 1]
```diff
@@ -33,9 +34,8 @@ public void tearDown() throws Exception {
         waitForPatientDeletion(patient.uuid);
     }
 
-    // Test for Story RA-71
     @Test
-    @Category(org.openmrs.reference.groups.BuildTests.class)
+    @Category(BuildTests.class)
```
Comment (kdaud) [inline ui-tests/src/test/java/org/openmrs/reference/RegisterPatientTest.java:38]: Refactored `@Category` annotation to align with the legacy ui test `convention`.

File: ui-tests/src/test/java/org/openmrs/reference/RegisterPatientTest.java [hunk 2]
```diff
@@ -33,10 +34,9 @@ public void tearDown() throws Exception {
         waitForPatientDeletion(patient.uuid);
     }
 
-    // Test for Story RA-71
     @Test
-    @Category(org.openmrs.reference.groups.BuildTests.class)
-    public void registerPatient() throws InterruptedException {
+    @Category(BuildTests.class)
```
Comment (kdaud) [inline ui-tests/src/test/java/org/openmrs/reference/RegisterPatientTest.java:39]: Refactoring `@Category` annotation to align with legacy ui test` convention`

File: ui-tests/src/test/java/org/openmrs/reference/RegisterPatientTest.java [hunk 3]
```diff
@@ -60,6 +61,4 @@ public void registerPatient() throws InterruptedException {
         assertThat(dashboardPage.getPatientGivenName(), is(patient.givenName));
         assertThat(dashboardPage.getPatientFamilyName(), is(patient.familyName));
     }
-
```
Comment (sherrif10) [inline ui-tests/src/test/java/org/openmrs/reference/RegisterPatientTest.java:63]: Remove these lines incase you dont need them for proper /openmrs javaconventions

Comment (kdaud) [reply ui-tests/src/test/java/org/openmrs/reference/RegisterPatientTest.java:63]: > Remove these lines incase you dont need them for proper /openmrs javaconventions

Thanks @sherrif10 for the review. You may need to take a look at the second commit with id `ff8904d`, it was already catered for within it.

File: ui-tests/src/test/java/org/openmrs/reference/RegisterPatientTest.java [hunk 4]
```diff
@@ -33,10 +35,9 @@ public void tearDown() throws Exception {
         waitForPatientDeletion(patient.uuid);
     }
 
-    // Test for Story RA-71
     @Test
-    @Category(org.openmrs.reference.groups.BuildTests.class)
-    public void registerPatient() throws InterruptedException {
+    @Category(BuildTests.class)
+    public void registerPatientTest() throws InterruptedException {
```
Comment (sherrif10) [inline ui-tests/src/test/java/org/openmrs/reference/RegisterPatientTest.java:40]: Any reason why you didnot remove interruptedException here ?

Comment (sherrif10) [inline ui-tests/src/test/java/org/openmrs/reference/RegisterPatientTest.java:40]: Any reason why you didnot remove interruptedException here ?

Comment (kdaud) [reply ui-tests/src/test/java/org/openmrs/reference/RegisterPatientTest.java:40]: > Any reason why you didnot remove interruptedException here ?

`Exception` is very important here to handle. When a` test method` is holding `function(s)` with cases where exception is expected within them, then by default the hosting method has to extend that specific `Exception`.

Comment (kdaud) [reply ui-tests/src/test/java/org/openmrs/reference/RegisterPatientTest.java:40]: > Any reason why you didnot remove interruptedException here ?

Made a response on this above.

**Pull request merged**